src/name/verify.rs

```diff
@@ -160,6 +157,10 @@ fn check_presented_id_conforms_to_constraints_in_subtree(
                 dns_name::presented_id_matches_constraint(name, base).ok_or(Error::BadDER)
             }
 
+            (GeneralName::DirectoryName(name), GeneralName::DnsName(base)) => {
+                common_name(name).map(|cn| cn == base)
```

Comment:
isn't `base` here a _constraint_ rather than a _name_? ie it should match if it is equal (as currently) but _also_ match (say) if the name is "www.foo.com" and the constraint allows ".foo.com"?

Reply:
i think my conclusion on this is: it's an excellent start, and fixes web-platform-tests's use of name constraints to reduce the strength of their testing CA. but for me it's underlining a big testing debt here. i will look at that in the coming days; especially and also in the effort to land briansmith/webpki#131